src/lib/swapper/swappers/OneInchSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -24,88 +25,89 @@ export async function getTradeQuote(
     chainId,
     sellAsset,
     buyAsset,
-    sellAmountBeforeFeesCryptoBaseUnit,
     accountNumber,
     affiliateBps,
+    eip1559Support,
+    receiveAddress,
   } = input
+  const sellAmount = input.sellAmountBeforeFeesCryptoBaseUnit
 
-  if (sellAmountBeforeFeesCryptoBaseUnit === '0') {
-    return Err(
-      makeSwapErrorRight({
-        message: '[getTradeQuote] sellAmountBeforeFeesCryptoBaseUnit must be greater than 0',
-        code: SwapErrorType.TRADE_BELOW_MINIMUM,
-      }),
-    )
-  }
+  const assertion = assertValidTrade({ buyAsset, sellAsset, receiveAddress })
+  if (assertion.isErr()) return Err(assertion.unwrapErr())
 
-  const { assetReference: fromAssetAddress } = fromAssetId(sellAsset.assetId)
-  const { assetReference: toAssetAddress } = fromAssetId(buyAsset.assetId)
+  const minimumCryptoHuman = getMinimumCryptoHuman()
+  const minimumCryptoBaseUnit = toBaseUnit(minimumCryptoHuman, sellAsset.precision)
 
-  const buyTokenPercentageFee = convertBasisPointsToPercentage(affiliateBps).toNumber()
+  const normalizedSellAmount = normalizeAmount(
+    bnOrZero(sellAmount).eq(0) ? minimumCryptoBaseUnit : sellAmount,
+  )
 
-  const apiInput: OneInchQuoteApiInput = {
-    fromTokenAddress: fromAssetAddress,
-    toTokenAddress: toAssetAddress,
-    amount: sellAmountBeforeFeesCryptoBaseUnit,
-    fee: buyTokenPercentageFee,
-  }
+  const buyTokenPercentageFee = convertBasisPointsToPercentage(affiliateBps).toNumber()
 
   const { chainReference } = fromChainId(chainId)
   const maybeQuoteResponse = await oneInchService.get<OneInchQuoteResponse>(
     `${apiUrl}/${chainReference}/quote`,
-    { params: apiInput },
+    {
+      params: {
+        fromTokenAddress: fromAssetId(sellAsset.assetId).assetReference,
+        toTokenAddress: fromAssetId(buyAsset.assetId).assetReference,
+        amount: normalizedSellAmount,
+        fee: buyTokenPercentageFee,
+      } as OneInchQuoteApiInput,
```

Comment:
Do we need the cast? tsc seems happy without

Reply:
you're right, was thinking it would give us type safety, but that's not how as works haha. might pull it back out into it's own variable